packages/frontend/src/features/post/hooks/useBackgroundReputationClaim/useBackgroundReputationClaim.ts

```diff
@@ -0,0 +1,196 @@
+                        currentEpoch,
+                        currentNonce,
+                        chainId,
+                        attesterId,
+                    })
+                return genProofAndVerify(
+                    UnirepSocialCircuit.reportNonNullifierProof,
+                    reportNonNullifierCircuitInput,
+                )
+            }
+        },
+        [],
+    )
+
+    const claimReputation = useMutation({
+        mutationKey: [MutationKeys.ClaimReputation],
+        mutationFn: async ({ report, repUserType }: ClaimReputationParams) => {
+            const userState = await getGuaranteedUserState()
+            const { publicSignals, proof } = await generateProof(
+                userState,
+                report,
+                repUserType,
+            )
+
+            const result = await relayClaimReputation(
+                report.reportId,
+                repUserType,
+                publicSignals,
+                flattenProof(proof),
+            )
+            await userState.waitForSync()
+
+            return {
+                ...result,
+                epoch: userState.sync.calcCurrentEpoch(),
+                epochKey: userState.getEpochKeys(),
+            }
+        },
+        onSuccess: (data) => {
+            queryClient.invalidateQueries({
+                queryKey: [QueryKeys.ReputationHistory],
+            })
+            queryClient.invalidateQueries({
+                queryKey: [QueryKeys.ReportHistory, data.reportId],
+            })
+            refetchReports()
+        },
+    })
+
+    const claimReputationInBackground = useCallback(
+        async (report: ReportHistory, repUserType: RepUserType) => {
+            try {
+                await claimReputation.mutateAsync({ report, repUserType })
+            } catch (error) {
+                console.error(
+                    'Failed to claim reputation in background:',
+                    error,
+                )
+            }
+        },
+        [claimReputation],
+    )
+
+    useEffect(() => {
+        const processReports = async () => {
+            if (!reportsWaitingForTransaction) return
+
+            const userState = await getGuaranteedUserState()
+
+            for (const report of reportsWaitingForTransaction) {
+                const epoch = report.reportEpoch
+                const epochKeyLiteProof = await userState.genEpochKeyLiteProof({
+                    epoch: epoch,
+                })
+                const currentEpochKey = epochKeyLiteProof.epochKey.toString()
+                if (
+                    report.reportorEpochKey === currentEpochKey &&
+                    !report.reportorClaimedRep
+                ) {
+                    await claimReputationInBackground(
+                        report,
+                        RepUserType.REPORTER,
+                    )
+                } else if (
+                    report.respondentEpochKey === currentEpochKey &&
```

Comment:
`respondentEpochKey` is the epochKey at the epoch of making a post, not `currentEpochKey`.